src/main/java/de/hysky/skyblocker/mixins/ClientPlayNetworkHandlerMixin.java

```diff
@@ -79,20 +78,20 @@ protected ClientPlayNetworkHandlerMixin(MinecraftClient client, ClientConnection
         return !Utils.isOnHypixel();
     }
 
-    @Inject(method = "onPlaySound", at = @At("RETURN"))
+    @Inject(method = "onPlaySound", at = @At("HEAD"), cancellable = true)
     private void skyblocker$onPlaySound(PlaySoundS2CPacket packet, CallbackInfo ci) {
         FishingHelper.onSound(packet);
         CrystalsChestHighlighter.onSound(packet);
-		RegistryEntry<SoundEvent> sound = packet.getSound();
+		SoundEvent sound = packet.getSound().value();
 
 		// Mute Enderman sounds in the End
 		if (Utils.isInTheEnd() && SkyblockerConfigManager.get().otherLocations.end.muteEndermanSounds) {
 			// Check if the sound identifier matches any Enderman sound identifiers
-			if (sound.matchesId(SoundEvents.ENTITY_ENDERMAN_AMBIENT.id()) ||
-					sound.matchesId(SoundEvents.ENTITY_ENDERMAN_DEATH.id()) ||
-					sound.matchesId(SoundEvents.ENTITY_ENDERMAN_HURT.id()) ||
-					sound.matchesId(SoundEvents.ENTITY_ENDERMAN_SCREAM.id()) ||
-					sound.matchesId(SoundEvents.ENTITY_ENDERMAN_STARE.id())) {
+			if (sound.id().equals(SoundEvents.ENTITY_ENDERMAN_AMBIENT.id()) ||
+					sound.id().equals(SoundEvents.ENTITY_ENDERMAN_DEATH.id()) ||
+					sound.id().equals(SoundEvents.ENTITY_ENDERMAN_HURT.id()) ||
+					sound.id().equals(SoundEvents.ENTITY_ENDERMAN_SCREAM.id()) ||
+					sound.id().equals(SoundEvents.ENTITY_ENDERMAN_STARE.id())) {
 				// Cancel the playback of Enderman sounds
 				ci.cancel();
 			}
```

Comment:
while testing "Mute Enderman sounds feature" i figured out that sound matchingid wasn't actually working (if statments always false) for FishingHelper and CrystalHollows Chests